### DISCUSSION TITLE
training vram consumption

### DISCUSSION THREAD

**Discussion by adrianpuiu** [seruva19/kubin#74] [Show and tell]
![Screenshot from 2023-05-17 20-36-50](https://github.com/seruva19/kubin/assets/187078/0b4e9cc9-c494-41d8-8eac-b227bcbc7347)

training 150 images / hour  @ 123 epochs

**Comment by seruva19** [comment]
Hmm.. So 16 Gb is enough?

**Reply by adrianpuiu** [reply]
yes, its more than enough, and i'm doing them cropped at 768px .
idk what the guy did with those dreambooth and textual inversion scripts but the vram consumption was way to much .. i mean i could have run the textual inversion one but  it was at 22  Gigs + constantly

**Reply by seruva19** [reply]
Interesting. Makes even more sense now to train prior and unclip parts sequentially.

**Reply by adrianpuiu** [reply]
![Screenshot from 2023-05-17 20-50-09](https://github.com/seruva19/kubin/assets/187078/a2de9931-3e71-4cec-acba-32561e80a014)
 yes this is what im doing.
1 by 1 
 btw , that above picture was for the unclip fp16 decoder , the prior training takes 11.7 Gigs.

**Reply by kodxana** [reply]
My benchmark on A6000 48 GB GPU:
For prio model I was able to get batch size 140 with 98% memoery usage.
For decoder model max batch size is 4 for 99% usage.

**Comment by seruva19** [comment]
Tested training on free tier colab, it works fine.
Batch size=1
Prior:  ~ 10.8 Gb
Unclip:  ~ 13.7 Gb

**Reply by adrianpuiu** [reply]
btw now that you mentioned colab  . theyll need a way to keep the models saved onto the google drive so they eliminate the models downloading times .

**Reply by seruva19** [reply]
There was already a checkbox in colab notebook for this ;)
As for saving trained checkpoints, you just have to set up correct paths like /content/gdrive/MyDrive/images etc. 
And (haven't tested it) perhaps it's possible to resume training later by using already trained checkpoint as source model.

**Reply by adrianpuiu** [reply]
@kodxana  we need your help man . check [this](https://github.com/seruva19/kubin/discussions/96) one out  .. the dude has already a 60k image data set from MJ 5 .. maybe you could help the guy with the training
